common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidPlatformUtil.java

```diff
@@ -213,6 +218,23 @@ public String getPackageNameFromUid(int uid) {
         return mContext.getPackageManager().getNameForUid(uid);
     }
 
+    @Override
+    public List<Map.Entry<String, String>> setPlatformSpecificExtraQueryParameters(@Nullable List<Map.Entry<String, String>> originalList) {
+        List<Map.Entry<String, String>> queryParams = originalList != null ?  new ArrayList<>(originalList) : new ArrayList<>();
+
+        // Passkey feature support is only for Android at the moment.
+        final  Map.Entry<String, String> webauthnParam = new AbstractMap.SimpleEntry<>(FidoConstants.WEBAUTHN_QUERY_PARAMETER_FIELD, FidoConstants.WEBAUTHN_QUERY_PARAMETER_VALUE);
+        if (CommonFlightManager.isFlightEnabled(CommonFlight.ENABLE_PASSKEY_FEATURE)) {
+            if (!queryParams.contains(webauthnParam)) {
+                queryParams.add(webauthnParam);
+            }
+        } else {
+            // If we don't want to add this query string param, then we should also remove other instances of it that might be already present from MSAL/OneAuth-MSAL.
```

Comment:
Does this mean we are overriding flights set by client apps?

Reply:
this is something even I was confused about, for the Linux case of supporting FIDO I checked with MSAL.CPP team, they do not pass the webauthn=1 parameter

Reply:
> Does this mean we are overriding flights set by client apps?

Yes. Sending webauthn=1 declares that your app supports/has done the necessary work to support passkeys. A brokered request should only send the parameter if the brokered apps themselves are ready (which we know well). So if a 1P app sends the parameter, which is declaring for itself, and it goes through broker, but broker isn't ready yet, we don't want to send that parameter to the server side.


> this is something even I am confused about, for the Linux case of supporting FIDO I checked with MSAL.CPP team, they do not pass the webauthn=1 parameter

OneAuth-MSAL Android has a configuration for sending the webauthn parameter, and the control is with the individual 1P apps. Linux on the MSAL.CPP side shouldn't have logic for sending this.